### DISCUSSION TITLE
Access to teammembers

### DISCUSSION THREAD

**Discussion by jkramps** [gwleuverink/clickup-time-tracker#10] [Ideas]
Access to other teammembers, is that doable? Can be restricted or open.

**Comment by gwleuverink** [comment]
I'm afraid that's a bit tricky. We use a personal token, which only allows you to fetch data attached to your account. 

So either we need to do multiple calls with multiple tokens or explore the oauth app approach (though i dont think it will work that way either)

Maybe nice to keep in mind for the future, but for now I don't want to overcomplicate things

**Reply by gwleuverink** [reply]
This seems impractical to implement in a generic and safe way. We could add a intermediary server to add more flexibility on the client side. But for now I'd rather keep things simple

**Reply by Patrick-56Bit** [reply]
@gwleuverink Actually the clickup API allows you to retrieve all time entries for any team members if the personal token belongs to a Clickup admin. The following API call will do this if you add the memberids to the Assignees query string:

https://jsapi.apiary.io/apis/clickup20/reference/0/time-tracking-20/get-time-entries-within-a-date-range.html

So a single api call can retrieve many time entries of many members (of the same workspace/teamid)

The memberids can't be retrieved programmatically unfortunately (unless one has an enterprise account). But its easy to add them manually into a list on your app. They can be retrieved by going to the Clickup "People" list.

Again, this is only possible if the personal token belongs to an admin or owner. From clickup api docs: _"Note: Only Workspace Owners/Admins have access to do this."_

Hope this helps.

**Reply by gwleuverink** [reply]
Interesting. That certainly opens up some possibilities. I'll think on this for a bit. It's definitely a nice feature to have
